base_classes/NXattenuator.nxdl.xml

```diff
@@ -43,9 +43,27 @@
         TODO  SHADOW: http://ftp.esrf.eu/pub/scisoft/shadow       or (older)   http://www.nanotech.wisc.edu/shadow  
     -->
 
-    <field name="distance" type="NX_FLOAT" units="NX_LENGTH">
+    <field name="distance" type="NX_FLOAT" units="NX_LENGTH" deprecated="Use transformations/NXtransformations/distance instead">
         <doc>Distance from sample</doc>
     </field>
+<field name="depends_on" type="NX_CHAR">
+  <doc>
+    NeXus positions components by applying a set of translations and rotations 
+    to apply to the component starting from 0, 0, 0. The order of these operations 
+    is critical and form a what NeXus calls a dependency chain. The depends_on 
+    field defines the path to the top most operation of the dependency chain. 
+    Usually these operations are stored in a NXtransformations group. But NeXus 
+    allows them to be stored anywhere.
+  </doc>
+</field>
+
+<group name="transformations" type="NXtransformations" minOccurs="0">
```

Comment:
Is this name fixed?

Reply:
I thought it is a good idea to have that in the same place in each group. Is there a use case for another name?

Reply:
The name is not important from a parsing perspective because you are always following a `depends_on` path to find it anyway. I also can't think of a use case for allowing a different name though.

Reply:
@prjemian Can you chime in on this?  From session G of CodeCamp2021, we are split on this issue.  Whether to specify the name as transformations or leave it optional.  (Note, nobody had a really strong opinion :)

Reply:
IMO, not sure why it is necessary to require the exact name.  Within the BES Bluesky communitym there is a group looking at XPCS data and has begun learning how to apply NX transformations.  They have assumed the name is not required to be exactly `transformations`.  Thus, in any NXDL spec, the name could be omitted unless there are two or more NXtransformations groups in the same directory.  In such case, they need names to avoid ambiguity.

Reply:
For the crystallography community, prescriptive naming is not likely to work.

On the matter of deprecating distance in favor of deriving it from the axis chain, that also is not likely to work, since distance, wavelength and beam center are essential data for
processing.  The best we are likely to be able t do is to make it clear when they are derived versus when they are directly observed and make derived the default.